crates/solver/src/driver.rs

```diff
@@ -435,13 +452,36 @@ impl Driver {
                     .map_err(|err| anyhow!("{err}"))
                     .context("convert nonce")?,
             };
+            let scores = model::solver_competition::Scores {
+                winner: winning_solver.account().address(),
```

Comment:
Double checking: Is it intended that the score you use here might come from a different settlement than winning_solver, winning_settlement` above? Because we don't guarantee that the scores and rated_settlements are sorted the same way.

Reply:
Came to ask the same thing, isn't the `winning_solver` not guaranteed to correspond with the winning score?

Reply:
Yes, it's a bug introduced when I added the winner. Thanks, fixed.